### PR TITLE
Remove Plug from strategies

### LINKED ISSUE #37: Detach plug from strategies
(From @yordis)

Methods in strategies are [currently expecting `Plug.Conn` argument](https://github.com/danschultzer/pow_assent/blob/master/lib/pow_assent/strategy.ex#L24-L29), but there is no need for this. The strategies should be completely detached from plug.

A caveat is how state is handled with OAuth2:

https://github.com/danschultzer/pow_assent/blob/89a52aa9dbfc9a71fe662e2c4df27a34bf645602/lib/pow_assent/strategies/oauth2.ex#L27-L36

It should be moved to plug or controller level:

https://github.com/danschultzer/pow_assent/blob/3e3dadd7c220c4d3b9bd148b24e4e06cab894730/lib/pow_assent/plug.ex#L26-L36
https://github.com/danschultzer/pow_assent/blob/89a52aa9dbfc9a71fe662e2c4df27a34bf645602/lib/pow_assent/phoenix/controllers/authorization_controller.ex#L29-L32

### PR DESCRIPTION
**Pull request by danschultzer**
Resolves #37 

This is a breaking change since the whole strategies API has been changed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -202,14 +202,14 @@ You can also use `PowAssent.Strategy`:
...
-  @spec authorize_url(Keyword.t(), Plug.Conn.t()) :: {:ok, %{conn: Plug.Conn.t(), url: binary()}} | {:error, any()}
-  def authorize_url(config, conn) do
+  @spec authorize_url(Keyword.t()) :: {:ok, %{url: binary()}} | {:error, term()}
+  def authorize_url(config) do
     # Generate authorization url
   end
 
-  @spec callback(Keyword.t(), Plug.Conn.t(), map()) :: {:ok, %{conn: Plug.Conn.t(), user: map()}} | {:error, any()}
+  @spec callback(Keyword.t(), map()) :: {:ok, %{user: map()}} | {:error, any()}
   def callback(config, conn, params) do
```
Comment (yordis) [inline README.md]: There is mismatch between the spec and the function (conn is not longer in the spec)

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,27 @@
...
+defmodule TestProvider do
+  @behaviour PowAssent.Strategy
+
+  @spec authorize_url(Keyword.t()) :: {:ok, %{url: binary()}} | {:error, term()}
+  def authorize_url(config) do
+    # Generate authorization url
+  end
+
+  @spec callback(Keyword.t(), map()) :: {:ok, %{user: map()}} | {:error, any()}
+  def callback(config, conn, params) do
```
Comment (yordis) [inline CHANGELOG.md]:
```suggestion
  def callback(config, params) do
```

File: README.md (hunk 1 above)
Comment (yordis) [inline README.md]: There is the mismatch between the spec and the function (conn is no longer in the spec)

```suggestion
  def callback(config, conn, params) do
```
```suggestion
  def callback(config, params) do
```

**Comment by danschultzer** [comment]
Thanks @yordis, amended fixes 😄

**Comment by danschultzer** [comment]
This PR is ready. I'll do a review tomorrow with fresh eyes, and merge to master.

**Pull request merged**